src/main/java/Duke.java

```diff
@@ -1,10 +1,103 @@
+import java.text.ParseException;
+import java.util.Scanner;
+
+/**
+ * Represents a Duke chatbot.
+ */
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Initiates a Duke object.
+     * @param filePath path in hard disk to store tasks
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    private void done(int num) {
```

Comment:
i feel like all these methods (done, delete, add) should be moved into the respective command classes instead